src/createSettingsFileTransformer.js

```diff
@@ -0,0 +1,162 @@
+var objectAssign = require('@adobe/reactor-object-assign');
+
+function isArrayReference(str) {
+  return isString(str) && str.indexOf('[') !== -1 && str.indexOf(']') !== -1;
+}
+function sanitizeArrayKey(pathStrSegment) {
+  return pathStrSegment.substr(
+    0,
+    // the name goes up to the start of the array bracket: 'someArrayName[]'
+    pathStrSegment.indexOf('[')
+  );
+}
+
+function isObject(value) {
+  if (value == null) {
+    return false;
+  }
+
+  return Boolean(
+    typeof value === 'object' &&
+      Object.prototype.toString.call(value) === '[object Object]' &&
+      !Array.isArray(value)
+  );
+}
+
+function isString(value) {
+  var type = typeof value;
+  return Boolean(
+    type === 'string' ||
+      (type === 'object' &&
+        value != null &&
+        !Array.isArray(value) &&
+        Object.prototype.toString.call(value) === '[object String]')
+  );
+}
+
+function traverseIntoSettings(
+  filePathString,
+  settings,
+  decorateWithDynamicHost
+) {
+  // nothing to do
+  if (
+    !isString(filePathString) ||
+    !filePathString.length ||
+    !isObject(settings)
+  ) {
+    return;
+  }
+
+  // get the next key to observe
+  var pathSegments = filePathString.split('.');
+  var settingToReplace = pathSegments.pop();
+  if (!settingToReplace) {
```

Comment:
You don't have a test for this error. I would prefer to not throw an error and just return the untouched settings object if we get here. I think the only case when you would get into this situation would be if you have a path ending  with a dot ('a.b.'). And for this case we could change the turbine schemas to not accept transforms with paths that end with a dot.

Reply:
Yes, this was the exact case I had in mind, but I can remove it.